### PR TITLE
Fix/push utils

### PR DESCRIPTION
**Pull request by iskdrews**
## Explanation

Pushing utils script. 

## Pre-Merge Checklist

- [ ] PR template is filled out
- [ ] **IF** this PR fixes a bug, a test that _would have_ caught the bug has been added
- [ ] PR is linked to the appropriate GitHub issue
- [ ] PR has been added to the appropriate release Milestone

### + If there are functional changes:

- [ ] Manual testing complete & passed
- [ ] "Extension QA Board" label has been applied

> PR template source from [github.com/MetaMask](https://github.com/MetaMask)

### CODE REVIEW

File: src/background/shared/utils.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * @property {() => void} resolve - A function that resolves the Promise.
+ * @property {() => void} reject - A function that rejects the Promise.
+ */
+
+/**
+ * Create a defered Promise.
+ *
+ * @returns {DeferredPromise} A deferred Promise.
+ */
+export function deferredPromise() {
```
Comment (0xmad) [inline src/background/shared/utils.ts:19]: Let's add return type here

**Review by 0xmad — COMMENTED** [review]
Just an improvement for ts :)

File: src/background/shared/utils.ts (hunk 1 above)
Comment (iskdrews) [reply src/background/shared/utils.ts]: Fixed.

File: src/types/index.ts [hunk 2]
```diff
@@ -135,4 +135,11 @@ export type ReduxAction = {
     payload?: any
     error?: boolean
     meta?: any
-}
\ No newline at end of file
+}
+
+export interface DeferredPromise<T> {
+    promise: Promise<T>;
+    resolve: T;
```
Comment (0xmad) [inline src/types/index.ts:142]: I think it should be `(arg: T) => void` for resolve and `(err: Error) => void` for reject.

**Review by 0xmad — APPROVED** [review]

**Pull request merged**